src/views/home/components/transaction-form/transaction-form.view.tsx

```diff
@@ -28,43 +27,47 @@ const TransactionForm: FC = () => {
 
   const onChainFromButtonClick = (from: Chain) => {
     setLocalTransaction({ ...localTransaction, from });
-    closeModal();
+    onOpenList("");
   };
   const onChainToButtonClick = (to: Chain) => {
     setLocalTransaction({ ...localTransaction, to });
-    closeModal();
+    onOpenList("");
   };
   const onTokenClick = (token: Token) => {
     setLocalTransaction({ ...localTransaction, token });
-    closeModal();
+    onOpenList("");
   };
-  const componentFromChain = () => (
-    <List
-      placeholder="Search Network"
-      list={{ type: "chain", items: chains, onClick: onChainFromButtonClick }}
-    />
-  );
-  const componentToChain = () => (
-    <List
-      placeholder="Search Network"
-      list={{ type: "chain", items: chains, onClick: onChainToButtonClick }}
-    />
-  );
-  const componentToken = () => (
-    <List
-      placeholder="Search token"
-      list={{ type: "token", items: tokens, onClick: onTokenClick }}
-    />
-  );
-  const onChainFromSelectorClick = () => {
-    openModal(componentFromChain);
-  };
-  const onChainToSelectorClick = () => {
-    openModal(componentToChain);
-  };
-  const onTokenSelectorClick = () => {
-    openModal(componentToken);
+  const showModal = () => {
```

Comment:
This is not an action, but a component instead:

`const Modal = () => { //... }`